### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by matgabriel**
2 changes:
- the possibility to mock service discovery for external systems on a given endpoint (e.g. embedded etcd)
- opentracing span information automatically forwarded from client to server on gRPC unary calls

### CODE REVIEW

**Comment by matgabriel** [comment]
> Would be good to have unit tests serving as examples of how to use the mocking.

I added the unit test sd_mock_test.go and simplified a bit the mocking design

File: sd.go [hunk 1]
```diff
@@ -181,43 +194,74 @@ func (m MockedRegistrationHandle) DeRegister(ctx context.Context) error {
...
 	}
-	Sugar.Infof("Service %s with tag %s mocked to local gRPC server on port %d", serviceName, tag, m.g.GrpcPort())
+	Sugar.Debugf("Service %s with tag %s mocked to local gRPC server on port %d", serviceName, tag, m.g.GrpcPort())
 	return []ServiceDefinition{result}, nil
 }
 
-func (m *MockedServiceDiscoveryToLocalGrpcServer) ResolveTags(tag string) (map[string][]ServiceDefinition, error) {
+func contains(slice []string, e string) bool {
+	for _, i := range slice {
+		return i == e
```
Comment (pdaures) [inline sd.go]: to fix

File: trace.go [hunk 2]
```diff
@@ -157,3 +163,93 @@ func Trace(span opentracing.Span, fields ...zlog.Field) {
...
+	if !ok {
+		md = metadata.New(nil)
+	} else {
+		md = md.Copy() //mandated by the doc
+	}
+	mdWriter := metadataRW{md}
+
+	err := span.Tracer().Inject(span.Context(), opentracing.HTTPHeaders, mdWriter)
+	if err != nil {
+		span.LogFields(zlog.String("error", "failed injecting span context in metadata"), zlog.Error(err))
```
Comment (pdaures) [inline trace.go]: put in gorillaz log?

File: sd.go (hunk 1 above)
Comment (matgabriel) [reply sd.go]: :+1:

File: trace.go (hunk 2 above)
Comment (matgabriel) [reply trace.go]: :+1:

**Pull request merged**